pkg/provider/aws/secret.go

```diff
@@ -0,0 +1,19 @@
+package aws
+
+import (
+	"github.com/klothoplatform/klotho/pkg/core"
+	"github.com/klothoplatform/klotho/pkg/provider/aws/resources"
+)
+
+func (a *AWS) GenerateSecretsResources(construct *core.Secrets, result *core.ConstructGraph, dag *core.ResourceGraph) error {
+	for _, single := range construct.Secrets {
+		secret := resources.NewSecret(construct.Provenance(), single, a.Config.AppName)
+		dag.AddResource(secret)
+		a.MapResourceDirectlyToConstruct(secret, construct)
+
+		secretVersion := resources.NewSecretVersion(secret, single)
```

Comment:
Dont we need to create a policy to access the secret version as well?

Reply:
ah yeah, thanks for the reminder. (When I first wrote this, the policy stuff wasn't in yet, I think)

Reply:
👍 yeah other than this, looks good to me